tests/network-tests/src/fixtures/membership/VerifyValidatorMembershipFixture.ts

```diff
@@ -49,22 +50,38 @@ export class VerifyValidatorMembershipFixture extends WithMembershipWorkersFixtu
   }
 
   private assertQueriedMembershipsAreValid(qMembers: MembershipFieldsFragment[]): void {
-    this.events.map((e, i) => {
+    this.events.forEach((e, i) => {
       const verification = this.verifications[i]
       if (verification.expectFailure) return
 
       const qMembership = qMembers.find((p) => p.id === verification.memberId.toString())
       Utils.assert(qMembership, 'Query node: Membership not found')
+
+      const { id, metadata } = qMembership
+      const isVerified = verification.isVerified
+      const expectedOutcome = isVerified ? 'verified' : 'unverified'
+      assert.equal(metadata.isVerifiedValidator, isVerified, `Member ${id} was not ${expectedOutcome}.`)
     })
   }
 
   protected assertQueryNodeEventIsValid(qEvent: AnyQueryNodeEvent, i: number): void {
-    // TODO: implement QN checks
+    // NOTE: These transactions do not create any QN events
   }
 
   async runQueryNodeChecks(): Promise<void> {
     await super.runQueryNodeChecks()
-    const qMembers = await this.query.getMembersByIds(this.verifications.map((m) => m.memberId))
-    this.assertQueriedMembershipsAreValid(qMembers)
+
+    const memberIds = this.verifications.reduce(
+      (ids, { memberId }) => (ids.some((id) => memberId.eq(id)) ? ids : [...ids, memberId]),
+      [] as MemberId[]
+    )
```

Comment:
Don't you think following would be much simpler 
```suggestion
    const memberIds = [...new Set(this.verifications.map((v) => v.memberId))]
```

Reply:
:100: